.github/workflows/test.yml

```diff
@@ -0,0 +1,46 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python package
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox
+    - name: Cache tox environments
+      id: cache-tox
+      uses: actions/cache@v1
```

Comment:
Setup-Python already handle caching: https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies

Reply:
(Will this still work out of the box for tox though?)

Reply:
I'm not sure, to be honest.

Reply:
The setup-python action doesn't cache the .tox folder.

Reply:
This makes sense as long as tox is kept :)

Reply:
In fact, I think we should pull out tox and rely on GitHub workflows for the "matrix" of different python versions (which is primarily what it is used for). Right now (after the merge of this PR) we test multiple python version multiple times (because tox matrix is run for each workflow python version) which is unnecessary and misleading if there is a failure, see: https://github.com/andhus/scantree/actions/runs/7415433113/job/20178552710#step:6:32 and compare with job name :/